api/pkg/apis/v1alpha1/managers/activations/activations-manager.go

```diff
@@ -216,6 +216,7 @@ func (t *ActivationsManager) ReportStatus(ctx context.Context, name string, name
 	}
 
 	current.UpdateTime = time.Now().Format(time.RFC3339) // TODO: is this correct? Shouldn't it be reported?
+	activationState.Spec = nil
```

Comment:
This will set the activationState.Spec to nil. It is not a good way to skip updating spec field given we support multiple state providers.